### PR TITLE
Remove "changes" from release note sections

### PR DESCRIPTION
**Pull request by electrum**
This makes the contents sidebar look better.

## Old
Screen Shot 2021-01-13 at 11 50 17 AM

## New: 
Screen Shot 2021-01-13 at 11 50 31 AM

### CODE REVIEW

File: docs/src/main/sphinx/release/release-322.rst [hunk 1]
```diff
@@ -2,19 +2,19 @@
...
-Server RPM changes
-------------------
+Server RPM
+----------
 
 * Fix a regression that caused zero-length files in the RPM. (:issue:`1767`)
 
-Connector changes
------------------
+Connector
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-322.rst:16]: nit: may be `Other Connectors` ?

**Review by phd3 — APPROVED** [review]
Nice, makes sense to me.

File: docs/src/main/sphinx/release/release-307.rst [hunk 2]
```diff
@@ -14,76 +14,76 @@ General changes
...
   bucketing version that uses an incompatible hash function. The Hive connector
   will treat such tables as not bucketed when reading and disallows writing. (:issue:`512`)
 * Add support for setting table comments via the :doc:`/sql/comment` syntax. (:issue:`200`)
 
-MySQL connector changes
------------------------
+MySQL connector
+---------------
 
 See `Base-JDBC connector library changes <#base-jdbc-connector-library-changes>`__.
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-307.rst:59]: these links need to be updated too

**Review by phd3 — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-0.148.rst [hunk 3]
```diff
@@ -90,29 +90,29 @@ Hive changes
...
-MySQL and PostgreSQL changes
-----------------------------
+MySQL and PostgreSQL
+--------------------
 
 * Push down predicates for ``VARCHAR``, ``DATE``, ``TIME`` and ``TIMESTAMP`` types.
 
-Other connector changes
------------------------
+Other connector
```
[the remark this answers is not in the document]
Comment (electrum) [reply docs/src/main/sphinx/release/release-0.148.rst:114]: Good catch! Fixed

File: docs/src/main/sphinx/release/release-322.rst (hunk 1 above)
Comment (electrum) [reply docs/src/main/sphinx/release/release-322.rst:16]: This was inconsistent, updated to match. I couldn't find any other instances like this

File: docs/src/main/sphinx/release/release-307.rst (hunk 2 above)
Comment (electrum) [reply docs/src/main/sphinx/release/release-307.rst:59]: Ah, I missed this one. I'll update it to the newer "Other connector" language.

**Comment by electrum** [comment]
Thanks for the careful review. I fixed those issues.

**Review by phd3 — APPROVED** [review]

**Pull request merged**